infra/core/env/event-hub/event-hub-secret-rbac.bicep

```diff
@@ -0,0 +1,38 @@
+
+@description('Required. The parameter object for eventHub. The object must contain the name, eventHubName, eventHubConnectionSecretName and resourceGroup values.')
+param eventHubNamespace object
+
+@description('Required. KeyVault name.')
+param keyVaultName string
+
+@description('Required. App Configuration Managed Identity Object id.')
+param appConfigMiObjectId string
+
+var roleDefinitionId = '4633458b-17de-408a-b874-0445c86b69e6' // Key Vault Secrets User
+
+resource keyVault 'Microsoft.KeyVault/vaults@2021-10-01' existing = {
+  name: keyVaultName
+}
+
+resource eventHubPolicy 'Microsoft.EventHub/namespaces/eventhubs/authorizationRules@2022-01-01-preview' existing = {
+  name: '${eventHubNamespace.name}/${eventHubNamespace.eventHubName}/FluxSendAccess'
+  scope: resourceGroup(eventHubNamespace.resourceGroup)
+}
+
+resource secretResource 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  parent: keyVault
+  name: eventHubNamespace.eventHubConnectionSecretName
+  properties: {
+    value: eventHubPolicy.listkeys().primaryConnectionString
```

Comment:
con string - we should use MI. Do we need to use connection string?

Reply:
@DaRoszko - No it's not possible - I did discuss with @kbabigumira.

Reply:
@DaRoszko  from what we have researched so far, we can't use MI because the token will expire. This will require further investigation as part of the next user story.